front-end/src/visualization/Visualization.jsx

```diff
@@ -57,20 +79,86 @@ function Visualization() {
         return positions;
     }
 
+    function getColorForValence(valence) {
+        const colors = [
+            { valence: 0.0, color: '#FF0000' }, // Red for very angry or extremely sad
+            { valence: 0.25, color: '#800080' }, // Purple for sad
+            { valence: 0.5, color: '#0000FF' }, // Blue for neutral or melancholic
+            { valence: 0.75, color: '#FFA500' }, // Orange for cheerful
+            { valence: 1.0, color: '#CCCC00' }  // Yellow for very happy
+        ];
+
+        let lower = colors[0];
+        let upper = colors[colors.length - 1];
+
+        for (let i = 0; i < colors.length; i++) {
+            if (colors[i].valence <= valence) {
+                lower = colors[i];
+            }
+            if (colors[i].valence >= valence) {
+                upper = colors[i];
+                break;
+            }
+        }
+
+        if (lower.valence === upper.valence) {
+            return upper.color;
+        }
```

Comment:
I wonder if this check is needed. This will happen for all cases where the valence is equal to the value inside the `color` constant. Can't we avoid the for loop at all if that condition is met?